lib/hyacinth/adapters/digital_object_search_adapter/solr/document_adapter.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module Hyacinth
+  module Adapters
+    module DigitalObjectSearchAdapter
+      class Solr::DocumentAdapter
+        def solr_document_for(digital_object)
+          solr_document = merge_core_fields_for(digital_object)
+          merge_dynamic_fields_for!(digital_object, solr_document)
+          merge_rights_fields_for!(digital_object, solr_document)
+          solr_document
+        end
+
+        def merge_core_fields_for(digital_object, solr_document = {})
+          merge_core_fields_for!(digital_object, solr_document.dup)
+        end
+
+        def merge_core_fields_for!(digital_object, solr_document = {})
+          indexable_title = ::Types::DigitalObjectInterface.title_for(digital_object, true)
+          solr_document.merge!(
+            'id' => digital_object.uid,
+            'digital_object_type_ssi' => digital_object.digital_object_type,
+            'uid_ssi' => digital_object.uid,
```

Comment:
Maybe not needed, since we have the value in `id`? (I realize this is a ported thing and not a new thing.)

Reply:
Also, there's no test checking for its presence.